mhs/common/mhs_common/workflow/asynchronous_express.py

```diff
@@ -66,7 +66,12 @@ def __init__(self, party_key: str = None, persistence_store: persistence_adaptor
         try:
             url = interaction_details['url']
             response = await self.transmission.make_request(url, http_headers, message)
-            ebxml_handler.handle_ebxml_error(response.code, response.headers, response.body)
+
+            code, body = ebxml_handler.handle_ebxml_error(response.code, response.headers, str(response.body))
+            if code == 500:
+                await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_TRANSMISSION_FAILED)
+                return code, body
```

Comment:
Can you log here and add a test for this scenario.

Reply:
Changed.